src/main/java/ch/njol/skript/expressions/ExprNearestEntity.java

```diff
@@ -0,0 +1,127 @@
+/**
+ * This file is part of Skript.
+ *
+ * Skript is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Skript is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.Literal;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.util.Kleenean;
+import ch.njol.util.StringUtils;
+import org.bukkit.Location;
+import org.bukkit.entity.Entity;
+import ch.njol.skript.lang.util.SimpleExpression;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.Arrays;
+
+@Name("Nearest Entity")
+@Description("Gets the entity nearest to a location or another entity.")
+@Examples({"kill the nearest pig and cow relative to player",
+	"teleport player to the nearest cow relative to player",
+	"teleport player to the nearest entity relative to player",
+	"on click:\n\tkill nearest pig"})
+@Since("INSERT VERSION")
+public class ExprNearestEntity extends SimpleExpression<Entity> {
+
+	static {
+		Skript.registerExpression(ExprNearestEntity.class, Entity.class, ExpressionType.SIMPLE,
+			"[the] nearest %*entitydatas% [[relative] to %entity/location%]",
+			"[the] %*entitydatas% nearest [to %entity/location%]");
+	}
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	@NonNull
+	private EntityData<?>[] entityDatas;
+
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	@NonNull
```

Comment:
@SuppressWarnings("NotNullFieldNotInitialized")
@NonNull

We don't need the NonNull here. Well actually we don't need either, but Intellij likes to warn that it's not initalized when we know it will always, and Intellij is probably still warning after you added the NonNull because it can't read the NonNull as the NonNull is a ECJ Eclipse annotation. So we can keep the warning annotation, but the NonNull isn't doing anything, even for Intellij users.

Reply:
All methods/fields/etc are NonNull by default 🙂

Reply:
This still needs to be addressed @Pikachu920

Reply:
removed the nonnull annotations 🥳